parser/prism/Translator.cc

```diff
@@ -200,6 +200,20 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
                 args.emplace_back(std::move(blockPassNode));
             }
 
+            // Handle safe navigation operator
+            auto callOperatorLoc = &callNode->call_operator_loc;
+            std::string_view callOperator;
+            if (callOperatorLoc != nullptr) {
+                callOperator = std::string_view(reinterpret_cast<const char *>(callOperatorLoc->start),
+                                                callOperatorLoc->end - callOperatorLoc->start);
+            }
+
+            if (callOperator == "&.") {
+                return make_unique<parser::CSend>(parser.translateLocation(loc), std::move(receiver),
```

Comment:
This early return causes the `translateCallWithBlock` call to be skipped below.

So this would have the wrong result for: e.g. `a&.map { }`. It'll behave as if the block wasn't there at all